### PR TITLE
Move P1 & P2 informations to seperate sensors and reduce code repitition

### PR DESCRIPTION
**Pull request by CM000n**
Hello, with reference to my issue https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/issues/2, in this PR, independent sensors are created from the information of P1 & P2 and these are removed from the attributes.

I have also simplified the code a little because it contained repetitive patterns, especially in the `async_update` methods of the sensor classes. So I created a common method for updating sensor data in the BaseSensor class to reduce repetition and utilized inheritance to set class attributes in `BasePowerSensor` and `BaseEnergySensor` instead of repeating them in each subclass.

I think the code can be further simplified by restructuring and using `SensorEntityDescription` instead of defining each sensor in it's own subclass. But I have to think more about that first

### CODE REVIEW

**Comment by mawoka-myblock** [comment]
have a look at this commit ;)
https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/commit/f06703e73c57e96909bf5e309c05f427c0141a57

**Review by mawoka-myblock — CHANGES_REQUESTED** [review]
Please remove the `Optional` again and use `TYPE | None` instead

**Comment by CM000n** [comment]
> Please remove the `Optional` again and use `TYPE | None` instead

If you really want to and this is the codestyle. But in my opinion, the `Optional` specification is much easier to read.

**Comment by CM000n** [comment]
> have a look at this commit ;) [f06703e](https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/commit/f06703e73c57e96909bf5e309c05f427c0141a57)

OK, what was the decision behind this? As I wrote in the issue, it's not really good style for Home Assistant sensors to put things that change frequently in the attributes.

**Comment by mawoka-myblock** [comment]
> > have a look at this commit ;) [f06703e](https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/commit/f06703e73c57e96909bf5e309c05f427c0141a57)
> 
> OK, what was the decision behind this? As I wrote in the issue, it's not really good style for Home Assistant sensors to put things that change frequently in the attributes.

I've done that, since I haven't read this part in the docs and I thought that this information was pretty niche

**Comment by CM000n** [comment]
> Please remove the `Optional` again and use `TYPE | None` instead

Reversed 😉

**Review by mawoka-myblock — APPROVED** [review]

**Comment by mawoka-myblock** [comment]
> > Please remove the `Optional` again and use `TYPE | None` instead
> 
> Reversed 😉

Thanks! I'm gonna have to test it tomorrow when the inverter is online ;)

**Comment by mawoka-myblock** [comment]
Sorry, I won't manage checking it out today. I'll try it tomorrow.

**Comment by lboue** [comment]
@mawoka-myblock Can you merge that PR?

**Review by mawoka-myblock — CHANGES_REQUESTED** [review]
Please remove all mentions of the attributes.

**Comment by CM000n** [comment]
> Please remove all mentions of the attributes.

Done (hopefully) 😃

File: custom_components/apsystemsapi_local/sensor.py [hunk 1]
```diff
@@ -138,16 +137,11 @@ def device_info(self) -> DeviceInfo:
...
-    def extra_state_attributes(self):
-        """Return entity specific state attributes."""
-        return self._attributes
-
     async def async_update_data(self):
         try:
             data = await self._api.get_output_data()
             self.update_state(data)
-            self._attr_available = True
+            self._attr_available = False
```
Comment (mawoka-myblock) [inline custom_components/apsystemsapi_local/sensor.py:144]: This "_attr_available" doesn't have anything to do with attributes. It just tells HA if the sensor is available or greyed out.

**Review by mawoka-myblock — CHANGES_REQUESTED** [review]
After that fix, you should be fine. Above was another mention of these available-thins.

Sorry for all the trouble.

File: custom_components/apsystemsapi_local/sensor.py (hunk 1 above)
Comment (CM000n) [reply custom_components/apsystemsapi_local/sensor.py]: Thx, didn't know. It's fixed now.

**Review by mawoka-myblock — APPROVED** [review]

**Comment by mawoka-myblock** [comment]
Finally (from my side)! Thank you for your patience and work! A new release will be created soon. Thanks! ❤️

**Comment by CM000n** [comment]
Thx for your helpfull feedback 😊

**Pull request merged**